File: Examen_Corto/Python/ejercicio7.py
import math
import psycopg2

numero = int(input('Ingrese un número: '))

if numero % 7 == 0:
    # Calcular el factorial si el número es divisible por 7
    factorial_resultado = math.factorial(numero)
    print(f'El factorial de {numero} es: {factorial_resultado}')

    # Almacenar el resultado en el archivo de texto
    nombre_archivo = 'ejercicio7.txt'
    with open(nombre_archivo, 'w') as file:
        file.write(f'El factorial de {numero} es: {factorial_resultado}\n')
    print(f'El resultado se ha guardado en {nombre_archivo}')

    # Almacenar el resultado en la variable 'salida'
    salida = str(factorial_resultado)

    # Conexión a la base de datos PostgreSQL
    try:
        conn = psycopg2.connect(
            dbname='ExamenCorto1',
            host='localhost',
            port='5432',
            user='postgres',
            password='gabrielgrdb'
        )

        cursor = conn.cursor()
        query = f"INSERT INTO salida VALUES ('{salida}')"
        cursor.execute(query)
        conn.commit()

        print('El resultado se ha insertado en la base de datos.')

    except psycopg2.Error as e:
        print(f'Error al insertar en la base de datos: {e}')

    finally:
        if conn:
            cursor.close()
            conn.close()

else:
    print('Error: El número no es divisible por 7.')

File: Examen_Corto/Octave/ejercicio7.m
numero = input('Ingrese un número: ');

if mod(numero, 7) == 0
    % Calcular el factorial si el número es divisible por 7
    factorialResultado = factorial(numero);
    disp(['El factorial de ' num2str(numero) ' es: ' num2str(factorialResultado)]);

    % Almacenar el resultado en el archivo de texto
    nombreArchivo = 'ejercicio7.txt';
    fid = fopen(nombreArchivo, 'w');
    if fid ~= -1
        fprintf(fid, 'El factorial de %d es: %d\n', numero, factorialResultado);
        fclose(fid);
        disp(['El resultado se ha guardado en ' nombreArchivo]);

        % Almacenar el resultado en la variable 'salida'
        salida = num2str(factorialResultado);  % Convertir a texto

        % Conexión a la base de datos PostgreSQL
        pkg load database % Cargar el paquete

        conn = pq_connect(setdbopts('dbname','ExamenCorto1','host','localhost',
            'port','5432','user','postgres','password','gabrielgrdb'));

        query = sprintf("INSERT INTO salida VALUES ('%s');", salida);
        N = pq_exec_params(conn, query);

        if ~isempty(N)
            disp('El resultado se ha insertado en la base de datos.');
        else
            disp('Error al insertar en la base de datos.');
        end

        pq_close(conn);
    else
        disp('Error al abrir el archivo para escribir.');
    end
else
    disp('Error: El número no es divisible por 7.');
end

